src/pocketmine/Player.php

```diff
@@ -2864,6 +2864,10 @@ public function handleDropItem(DropItemPacket $packet) : bool{
 			// Windows 10 Edition drops the contents of the crafting grid on container close - including air.
 			return true;
 		}
+		
+		if($this->gamemode === self::SPECTATOR) {
```

Comment:
It's not so much that it's difficult to read... It's the advantages of using an API that save trouble later.
1) If in the future there is a change in the spectator system, the isSpectator() function can be modified and the pmmp team won't have to update every non-api spectator check.
2) APIs make code look cleaner, and reduces boilerplate code. (Readability)
3) Reduces the chance of having bugs.

> I never said it was noticeable

> but what if a server has 9000 players (on 1 PM server not realistic but theoretical) and the function is run for each of them? Every micro optimization counts ;)

You implied your intension pretty well. Performance issues don't come from calling a function.. they are mostly caused by file I/O, and big loops running expensive operations every tick. No micro optimization counts.

”We should forget about small efficiencies, say about 97% of the time: premature optimization is the root of all evil.” ~  DonaldKnuth